fontir/src/orchestration.rs

```diff
@@ -45,89 +40,54 @@ pub struct Context {
     // a subset of the full input.
     pub input: Arc<Input>,
 
-    // If present, the one and only key you are allowed to write to
-    // Otherwise you totally get to write whatever you like
-    write_mask: Option<WorkIdentifier>,
-
-    // If present, what you can access through this context
-    // Intent is root has None, task-specific Context only allows access to dependencies
-    read_mask: Option<HashSet<WorkIdentifier>>,
+    acl: Acl<WorkIdentifier>,
 
     // work results we've completed or restored from disk
     // We create individual caches so we can return typed results from get fns
     static_metadata: Cache<Option<Arc<ir::StaticMetadata>>>,
     glyph_ir: Cache<HashMap<String, Arc<ir::Glyph>>>,
 }
 
-#[derive(Clone)]
-struct Cache<T: Default> {
-    item: Arc<RwLock<T>>,
-}
-
-impl<T: Default> Cache<T> {
-    fn new() -> Cache<T> {
-        Cache {
-            item: Default::default(),
+impl Context {
+    fn copy(&self, acl: Acl<WorkIdentifier>) -> Context {
```

Comment:
I'd maybe call this `with_acl` or something? `copy` is heavily associated with the `Copy` trait.

Reply:
Good catch, updated